robocop/config/config.py

```diff
@@ -45,22 +62,33 @@ def translate_patterns(self):
-        parser.add_argument('-r', '--reports', action=ParseDelimetedArgAction, default=self.reports,
-                            help='Run reports')
-        parser.add_argument('-f', '--format', type=str, help='Format of output message', default=self.format)
-        parser.add_argument('-c', '--configure', action=ParseCheckerConfig, default=self.configure,
-                            help="Configure checker with parameter value")
-        parser.add_argument('-o', '--output', type=argparse.FileType('w'), default=self.output,
-                            help='Path to output file')
-        parser.add_argument('--filetypes', action=ParseFileTypes, default=self.filetypes,
-                            help='Comma separated list of file extensions to be scanned by Robocop')
-        parser.add_argument('paths', metavar='paths', type=str, nargs='+',
-                            help='List of paths (files and directories) to be scanned by Robocop')
+        parser = argparse.ArgumentParser(prog='robocop',
+                                         description='Static code analysis tool for Robot Framework',
+                                         epilog='For full documentation visit: '
+                                                'https://github.com/bhirsz/robotframework-robocop',
+                                         add_help=False)
+        required = parser.add_argument_group(title='Required parameters')
+        optional = parser.add_argument_group(title='Optional parameters')
+
+        required.add_argument('paths', metavar='paths', type=str, nargs='+', help=self.HELP_MSGS['help_paths'])
+
+        optional.add_argument('-i', '--include', action=ParseDelimetedArgAction, default=self.include,
+                              help=self.HELP_MSGS['help_include'])
+        optional.add_argument('-e', '--exclude', action=ParseDelimetedArgAction, default=self.exclude,
+                              help=self.HELP_MSGS['help_exclude'])
+        optional.add_argument('-r', '--reports', action=ParseDelimetedArgAction, default=self.reports,
+                              help=self.HELP_MSGS['help_reports'])
+        optional.add_argument('-f', '--format', type=str, default=self.format, help=self.HELP_MSGS['help_format'])
+        optional.add_argument('-c', '--configure', action=ParseCheckerConfig, default=self.configure,
+                              help=self.HELP_MSGS['help_configure'])
+        optional.add_argument('-o', '--output', type=argparse.FileType('w'), default=self.output,
+                              help=self.HELP_MSGS['help_output'])
+        optional.add_argument('--filetypes', action=ParseFileTypes, default=self.filetypes,
+                              help=self.HELP_MSGS['help_filetypes'])
+        optional.add_argument('-h', '--help', action='help', help=self.HELP_MSGS['help_info'])
+        optional.add_argument('-v', '--version', action='version', version=__version__,
```

Comment:
what happens if I want to type this:
```
robocop -v
```
? Most of the python tools support only asking for version without need for supplying required argument. But we would need to change behaviour a bit. For example preparse options and leave program (after printing version).

Reply:
It works out of the box in argparse by using dedicated actions like `help` and `version`. It then ignores other parameters, prints the info and exit the application.